presto-hive/src/main/java/com/facebook/presto/hive/parquet/ParquetFileWriterFactory.java

```diff
@@ -96,6 +97,8 @@ public ParquetFileWriterFactory(
             return Optional.empty();
         }
 
+        CompressionCodecName compressionCodecName = getCompression(conf);
```

Comment:
Just curious,  can we read parquet compression directly using `HiveSessionProperties::getCompressionCodec(ConnectorSession session)`

Reply:
When I made the code change, the source of truth for compression is from config and the conf is set through ConfigurationUtils::configureCompression. Maybe your way would also work.